packages/flutter/test/material/dialog_test.dart

```diff
@@ -243,6 +243,32 @@ void main() {
     expect(await result, equals(42));
   });
 
+  testWidgets('Custom padding on SimpleDialogOption', (WidgetTester tester) async {
+    const EdgeInsets customPadding = EdgeInsets.fromLTRB(4, 10, 8, 6);
+    final SimpleDialog dialog = SimpleDialog(
+      title: const Text('Title'),
+      children: <Widget>[
+        SimpleDialogOption(
+          onPressed: () {},
+          child: const Text('First option'),
+          padding: customPadding,
+        ),
+      ],
+    );
+
+    await tester.pumpWidget(_buildAppWithDialog(dialog));
+    await tester.tap(find.text('X'));
+    await tester.pumpAndSettle();
+
+    final Rect dialogRect = tester.getRect(find.byType(SimpleDialogOption));
+    final Rect textRect = tester.getRect(find.text('First option'));
+
+    expect(textRect.left, dialogRect.left + customPadding.left);
+    expect(textRect.top, dialogRect.top + customPadding.top);
+    expect(textRect.right, dialogRect.right - customPadding.right);
+    expect(textRect.bottom, dialogRect.bottom - customPadding.bottom);
```

Comment:
@shihaohong, would you prefer the hardcoded values in the assertions?

Reply:
I don't have a preference, this looks good to me!